src/main/java/org/junit/rules/ExternalResource.java

```diff
@@ -77,8 +77,10 @@ protected void before() throws Throwable {
 
     /**
      * Override to tear down your specific external resource.
+     *
+     * @throws Exception if teardown fails
      */
-    protected void after() {
+    protected void after() throws Exception {
```

Comment:
While it's rare for methods to throw `Throwable`, when you call a method using reflection you often handle the `InvocationTargetException` by throwing the cause (a `Throwable`). That's why ` Statement.evaluate()` is declared to throw `Throwable`.

I don't think it harms anything to have the method declared to throw `Throwable`. The calling code catches `Throwable` and subclasses of `ExternalResource` can override the method to throw a different exception.